File: part2/services/simple/SimpleClient.java
import org.apache.thrift.TException;
import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.protocol.TProtocol;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.transport.TTransport;

public class SimpleClient {
    public static void main(String[] args) throws TException {
        TTransport trans = new TSocket("localhost", 9090);
        TProtocol proto = new TBinaryProtocol(trans);
        SocialLookup.Iface client = new SocialLookup.Client(proto);
    
        trans.open();
        System.out.println("Number 1 site: " + client.GetSiteByRank(1));
        System.out.println("Twitter rank : " + client.GetSiteRankByName("Twitter"));
        trans.close();
    }
}

File: part3/cpp/simple/simple_client.cpp
#include <iostream>
#include <boost/make_shared.hpp>
#include <boost/shared_ptr.hpp>
#include <thrift/transport/TSocket.h>
#include <thrift/transport/TBufferTransports.h>
#include <thrift/protocol/TBinaryProtocol.h>
#include "gen-cpp/Message.h"

using namespace apache::thrift::transport;
using namespace apache::thrift::protocol;

int main() {
    boost::shared_ptr<TTransport> trans;
    trans = boost::make_shared<TSocket>("localhost", 8585);
    trans = boost::make_shared<TBufferedTransport>(trans);
    auto proto = boost::make_shared<TBinaryProtocol>(trans);
    MessageClient client(proto);

    trans->open();
    std::string msg;
    for (auto i = 0; i < 3; ++i) {
        client.motd(msg);
        std::cout << msg << std::endl;
    }
    trans->close();
}
